src/core/interpreter.cc

```diff
@@ -899,133 +901,145 @@ int Interpreter::RedisGenericCommand(bool raise_error, bool async, ObjectExplore
    * to luaRedisGenericCommand(), which normally should never happen.
    * To make this function reentrant is futile and makes it slower, but
    * we should at least detect such a misuse, and abort. */
-  if (cmd_depth_) {
-    const char* recursion_warning =
-        "luaRedisGenericCommand() recursive call detected. "
-        "Are you doing funny stuff with Lua debug hooks?";
-    PushError(lua_, recursion_warning);
-    return 1;
-  }
-
-  if (!redis_func_) {
-    PushError(lua_, "internal error - redis function not defined");
-    return raise_error ? RaiseError(lua_) : 1;
-  }
-
-  cmd_depth_++;
-  int argc = lua_gettop(lua_);
+  {
```

Comment:
I see now. This is where scope starts and it ends at line 1040. 
Have you considered factoring out the whole code within the scope into another function?

Reply:
That's a good idea. Maybe have this function return an error or int, and the caller will call `lua_error()` if needed?

Reply:
I split the code RedisGenericCommand into smaller functions now will return optional in case of error
please take a look and let me know what you thing